src/ert/job_queue/driver.py

```diff
@@ -101,65 +106,130 @@ async def kill(self, realization: "RealizationState") -> None:
         realization.verify_kill()
 
 
+LSF_STATUSES = [
+    "PEND",
+    "SSUSP",
+    "PSUSP",
+    "USUSP",
+    "RUN",
+    "EXIT",
+    "ZOMBI",
+    "DONE",
+    "PDONE",
+    "UNKWN",
+]
+
+
 class LSFDriver(Driver):
     def __init__(self, queue_options: Optional[List[Tuple[str, str]]]):
         super().__init__(queue_options)
 
         self._realstate_to_lsfid: Dict["RealizationState", str] = {}
         self._lsfid_to_realstate: Dict[str, "RealizationState"] = {}
+        self._max_attempt = 100
+        self._MAX_ERROR_COUNT = 100
+        self._statuses = {}
         self._submit_processes: Dict[
             "RealizationState", "asyncio.subprocess.Process"
         ] = {}
-
+        self._SLEEP_PERIOD = 3
         self._currently_polling = False
 
     async def submit(self, realization: "RealizationState") -> None:
         submit_cmd: List[str] = [
-            "bsub",
+            self.get_option("BSUB_CMD") if self.has_option("BSUB_CMD") else "bsub",
             "-J",
             f"poly_{realization.realization.run_arg.iens}",
             str(realization.realization.job_script),
             str(realization.realization.run_arg.runpath),
         ]
-        process = await asyncio.create_subprocess_exec(
-            *submit_cmd,
-            stdout=asyncio.subprocess.PIPE,
-            stderr=asyncio.subprocess.PIPE,
-        )
-        self._submit_processes[realization] = process
-
-        # Wait for submit process to finish:
-        output, error = await process.communicate()
-        print(output)  # FLAKY ALERT, we seem to get empty
-        print(error)
+        current_attempt = 0
+        while current_attempt < self._max_attempt:
```

Comment:
Since this retry pattern is repeated for all three commands, they should be refactored to avoid duplicated lines of code.